pages/interface/components/Header/index.js

```diff
@@ -71,13 +72,8 @@ export default function HeaderComponent() {
         </PrimerHeader.Item>
       </Box>
 
-      {!isLoading && (
-        <PrimerHeader.Item
-          sx={{
-            display: user ? ['none', 'flex'] : 'flex',
-            ml: 3,
-            mr: [1, , 3],
-          }}>
+      {!isLoading && !(user && isScreenSmall) && (
```

Comment:
Você não acha que a expressão fica mais simples convertendo para `||`?

```suggestion
      {!isLoading && (!user || !isScreenSmall) && (
```

Reply:
Sobre entender a expressão, não vejo diferença, então posso mudar se acharem mais fácil. Mas eu não pensei em deixar claro o que a lógica faz, mas em porque existe essa lógica aí.

Acho um pouco mais simples de entender a razão do condicional se pensar em "ocultar o item no caso específico em que o usuário está logado e ao mesmo tempo usando tela pequena", pois é o caso em que podemos mover o botão de pesquisa para o menu. Pensei que poderia exigir um pouco mais de esforço para entender a razão dessa lógica ao pensar em "mostrar o item se o usuário não estiver logado ou se não estiver em tela pequena".

Pensando por esse lado, o que você acha mais fácil? Ou até mesmo invertendo como abaixo:

```suggestion
      {!isLoading && !(isScreenSmall && user) && (
```

Reply:
> "ocultar o item no caso específico em que o usuário está logado e ao mesmo tempo usando tela pequena"

Talvez seja falta de costume meu em ler condições assim, mas ficou claro o raciocínio. Não vejo problemas em deixar do jeito que está, `!(user && isScreenSmall)`.